chain/chain/src/types.rs

```diff
@@ -545,7 +546,17 @@ pub trait RuntimeAdapter: Send + Sync {
         prev_epoch_last_block_hash: &CryptoHash,
         epoch_id: &EpochId,
         next_epoch_id: &EpochId,
-    ) -> Result<(BlockInfo, BlockInfo, BlockInfo, EpochInfo, EpochInfo, EpochInfo), Error>;
+    ) -> Result<
+        (
+            Arc<BlockInfo>,
+            Arc<BlockInfo>,
+            Arc<BlockInfo>,
+            Arc<EpochInfo>,
+            Arc<EpochInfo>,
+            Arc<EpochInfo>,
```

Comment:
Yeah... I decided to not touch this for the time being -- I want to actually understand what the code is doing, maybe we can just refactor this away instead of just refactoring...